src/libraries/Common/src/System/Security/Cryptography/HashOneShotHelpers.cs

```diff
@@ -12,40 +12,18 @@ internal static class HashOneShotHelpers
     {
         internal static byte[] HashData(HashAlgorithmName hashAlgorithm, ReadOnlySpan<byte> source)
         {
-            if (hashAlgorithm == HashAlgorithmName.SHA256)
-            {
-                return SHA256.HashData(source);
-            }
-            else if (hashAlgorithm == HashAlgorithmName.SHA1)
-            {
-                return SHA1.HashData(source);
-            }
-            else if (hashAlgorithm == HashAlgorithmName.SHA512)
-            {
-                return SHA512.HashData(source);
-            }
-            else if (hashAlgorithm == HashAlgorithmName.SHA384)
-            {
-                return SHA384.HashData(source);
-            }
-            else if (hashAlgorithm == HashAlgorithmName.SHA3_256)
-            {
-                return SHA3_256.IsSupported ? SHA3_256.HashData(source) : throw new PlatformNotSupportedException();
-            }
-            else if (hashAlgorithm == HashAlgorithmName.SHA3_384)
-            {
-                return SHA3_384.IsSupported ? SHA3_384.HashData(source) : throw new PlatformNotSupportedException();
-            }
-            else if (hashAlgorithm == HashAlgorithmName.SHA3_512)
-            {
-                return SHA3_512.IsSupported ? SHA3_512.HashData(source) : throw new PlatformNotSupportedException();
-            }
-            else if (Helpers.HasMD5 && hashAlgorithm == HashAlgorithmName.MD5)
-            {
-                return MD5.HashData(source);
-            }
-
-            throw new CryptographicException(SR.Format(SR.Cryptography_UnknownHashAlgorithm, hashAlgorithm.Name));
+            return hashAlgorithm.Name switch
+            {
+                HashAlgorithmNames.SHA256 => SHA256.HashData(source),
+                HashAlgorithmNames.SHA1 => SHA1.HashData(source),
+                HashAlgorithmNames.SHA512 => SHA512.HashData(source),
+                HashAlgorithmNames.SHA384 => SHA384.HashData(source),
+                HashAlgorithmNames.SHA3_256 => SHA3_256.HashData(source),
```

Comment:
Rather than doing the IsSupported check in the caller, I guess it's now just relying on the check inside HashData?  If so, cool.

Reply:
Yep. Don't need to check it twice.